Public/Src/Cache/ContentStore/Distributed/Sessions/DistributedContentSession.cs

```diff
@@ -407,6 +422,24 @@ private Task<ProactiveCopyResult> RequestProactiveCopyIfNeededAsync(OperationCon
                 });
         }
 
+        private async Task<BoolResult> RequestOrPushContentAsync(OperationContext context, ContentHash hash, MachineLocation target, bool isInsideRing)
+        {
+            if (Settings.PushProactiveCopies)
+            {
+                var streamResult = await Inner.OpenStreamAsync(context, hash, context.Token);

```

Comment:
Need using around the stream to ensure disposal.

Reply:
Not sure about this. Wrapping on "using" here would mean that we would dispose it as soon as we return it, wouldn't it?

---
In reply to: [343831893](https://github.com/microsoft/BuildXL/pull/1141#discussion_r343831893) [](ancestors = 343831893)

Reply:
Sorry, you are right. I didn't see that this lambda returned a stream. So change to: please be sure to analyze when this stream will be disposed, including error & cancellation paths. I will look too, but your knowledge of this code is deeper.

---
In reply to: [343832763](https://github.com/microsoft/BuildXL/pull/1141#discussion_r343832763) [](ancestors = 343832763,343831893)